grpc/chunk/chunker.go

```diff
@@ -0,0 +1,107 @@
+// Package chunk provides a utility for sending sets of protobuf messages in
+// groups of smaller chunks. This is useful for gRPC, which has limitations around the maximum
+// size of a message that you can send.
+//
+// This code is adapted from the gitaly project, which is licensed
+// under the MIT license. A copy of that license text can be found at
+// https://mit-license.org/.
+//
+// The code this file was based off can be found here: https://gitlab.com/gitlab-org/gitaly/-/blob/v16.2.0/internal/helper/chunk/chunker.go
+package chunk
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// Message is a protobuf message.
+type Message interface {
+	proto.Message
+}
```

Comment:
just curious: why the redefinition?

Reply:
I haven't used generics much, but I remember when I introduced into sourcegraph/sourcegraph earlier, I remember getting compiler errors for writing expressions that looked like [T proto.Message]. I tried this again and the compiler didn't complain. Perhaps there was another syntax error earlier (or maybe Go's type inference has improved)?